Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -105,29 +105,74 @@ public IEnumerable<ParticipantStatusExtendedViewModel> GetParticipantStatusHisto
-                            .Join(context.Team.Where(t => t.StatusID != 0),
-                                pt => pt.TeamID,
-                                t => t.ID,
-                                (pt, t) => new TeamExtendedViewModel
-                                {
-                                    ID = t.ID,
-                                    Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
-                                    Name = t.Name,
-                                    Status = context.TeamStatus
-                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
-                                                .Description,
-                                    Logo = t.Logo,
-                                    TeamRecord = context.SeriesTeam
-                                        .Include(st => st.Team)
-                                        .Where(st => st.TeamID == t.ID)
-                                        .Select(st => (TeamRecordViewModel)st)
-                                
-                                });
+        var teams = context.ParticipantTeam
+                 .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                     .Join(context.Team.Where(t => t.StatusID != 0),
+                         pt => pt.TeamID,
+                         t => t.ID,
+                         (pt, t) => new TeamExtendedViewModel
+                         {
+                             ID = t.ID,
+                             Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                             Name = t.Name,
+                             Status = context.TeamStatus
+                                         .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                         .Description,
+                             Logo = t.Logo,
+                             TeamRecord = context.SeriesTeam
+                                 .Include(st => st.Team)
+                                 .Where(st => st.TeamID == t.ID)
+                                 .Select(st => (TeamRecordViewModel)st)
+
+                         });
         return teams;
     }
 
+    public IEnumerable<MatchExtendedViewModel> GetParticipantMatches(string username)
+    {
+        var matches = context.ParticipantTeam
+            .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                .Join(context.MatchTeam.Where(mt => mt.StatusID != 0 && mt.Match.StatusID != 0)
+                                        .Include(mt => mt.Match.Series.Activity)
+                                        .Include(mt => mt.Match.MatchTeam)
+                                                .ThenInclude(mt => mt.Match)
+                                        .Include(mt => mt.Match.MatchTeam)
+                                                .ThenInclude(mt => mt.Status),
```

Comment:
Since the `ThenIncludes` use the same root `Include`, I think you don't have to repeat it:
```suggestion
                                        .Include(mt => mt.Match.MatchTeam)
                                                .ThenInclude(mt => mt.Match)
                                                .ThenInclude(mt => mt.Status),
```

Reply:
The MatchTeam `Status` is a different table from the Match `Status`
If I did a ThenInclude on Match -> Status that would be different than MatchTeam -> Status

Reply:
Ah you're right. I keep thinking the scoping of ThenInclude should work differently then it does.